api/src/main/java/org/openmrs/module/pihcore/task/ClosePregnancyProgramTask.java

```diff
@@ -62,7 +64,21 @@ public void run() {
                                 Context.getProgramWorkflowService().savePatientProgram(patientPregnancyProgram);
                             } else if (currentState.getState().equals(antenatalState) && currentState.getStartDate().before(elevenMonthsAgo)) {
                                 patientPregnancyProgram.setDateCompleted(new Date());
-                                patientPregnancyProgram.setOutcome(lostToFollowup);
+                                patientPregnancyProgram.setOutcome(lostToFollowupConcept);
+                                Context.getProgramWorkflowService().savePatientProgram(patientPregnancyProgram);
+                            }
+                        }
+                    }
+                }
+
+                // now fetch all pregnancy programs that *have* been completed within the last year, and if their final state is miscarried, set the outcome to treatment complete
+                // note: when back entering information from more than a year ago, this will not work as expected, as for performance reasons it will only look at the last year
+                patientPregnancyPrograms = Context.getProgramWorkflowService().getPatientPrograms(null, pregnancyProgram, null, null, new DateTime().minusYears(1).toDate(), new Date(), true);
+                for (PatientProgram patientPregnancyProgram : patientPregnancyPrograms) {
+                    if (patientPregnancyProgram.getDateCompleted() != null && patientPregnancyProgram.getOutcome() == null) {
+                        for (PatientState patientState : patientPregnancyProgram.getMostRecentStateInEachWorkflow()) {
+                            if (patientState.getState().equals(miscarriedState)) {
```

Comment:
Probably don't need to check workflow here, but in case we do, figured I'd mention it

Reply:
Yep, nope... only one workflow for this program (and assumedly if there were more, miscarried would only be in one of them)